evalml/automl/automl_search.py

```diff
@@ -674,22 +680,14 @@ def _compute_cv_scores(self, pipeline, X, y):
                 logger.debug(f"\t\t\tFold {i}: {self.objective.name} score: {scores[self.objective.name]:.3f}")
                 score = scores[self.objective.name]
             except Exception as e:
+                if self.error_callback is not None:
+                    self.error_callback(e, self, fold_num=i, pipeline=pipeline)
```

Comment:
Cool, so because we pass `self` back in the callback, someone could theoretically write a callback which responds to failure and calls `automl.search` again. So this is implementing one part of an asynchronous API (#1047 ). Neat!

Reply:
Yes!!